### PR TITLE
Be compatible with standard UPDATE/DELETE syntax

### PR DESCRIPTION
**Pull request by etah000**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

Changelog category (leave one):
- Improvement

Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Support standard SQL syntax for UPDATE/DELETE statement.

Detailed description / Documentation draft:
Support the following UPDATE/DELETE statement:
* DELETE FROM db.name [ON CLUSTER cluster] [WHERE ...]
* UPDATE db.name [ON CLUSTER cluster] SET col_name = expr, ... [WHERE ...]

WHERE clause is optional and *db.name* is a non-distributed table.

### CODE REVIEW

**Review by alexey-milovidov — COMMENTED** [review]
We have initially decided to implement UPDATE/DELETE under ALTER syntax to highlight the difference - that these operations are very heavy and cannot be executed frequently.

But I know that there are use-cases where compatibility is bigger concern and looks like this PR addressed this concern.

Let's introduce a setting (in Settings.h) that will allow to explicitly enable/disable support for UPDATE/DELETE under compatible syntax. So it can be enabled for better compatibility or disabled for better user awareness.

Queries will always parse successfully but if this setting is disabled, ClickHouse should emit exception with clear explanation.

Let's make this setting disabled by default at least for the next release.

**Comment by alexey-milovidov** [comment]
Also let's do one of the following:
- make WHERE clause not optional for DELETE;
- introduce a setting that will forbid omition of WHERE clause in DELETE by default.

`DELETE FROM table` without WHERE is a common source of disasters.

**Comment by etah000** [comment]
Yes, update/delete are a resource consuming operations, especially when there is  a huge amount of data in DB.
A control setting is a good suggestion, I'll make the necessary changes.

For update/delete compatibilty, if we enforec where clause in the statements, it will not be compatiable with the standard SQL syntax, so we cannot claim that the clickhouse SQL is SQL 99 compatible. UPDATE/DELETE is frequently used in testing/development environment where the data volume is usually small, so those operations are processed quickly no matter whether  there is a "where clause" or not.

Of course,  Making "where clause" optional complicates the code logic. If you insist, I will delete related code changes.

**Comment by alexey-milovidov** [comment]
@etah000 Let's fix build check and style check.

File: src/Functions/FunctionsLogical.cpp [hunk 1]
```diff
@@ -42,7 +42,7 @@ using UInt8Container = ColumnUInt8::Container;
 using UInt8ColumnPtrs = std::vector<const ColumnUInt8 *>;
 
 
-MutableColumnPtr buildColumnFromTernaryData(const UInt8Container & ternary_data, const bool make_nullable)
+MutableColumnPtr convertFromTernaryData(const UInt8Container & ternary_data, const bool make_nullable)
```
Comment (alexey-milovidov) [inline src/Functions/FunctionsLogical.cpp:45]: You have accidentially commited unrelated file from old revision.

File: tests/queries/0_stateless/01379_with_fill_several_columns.sql [hunk 2]
```diff
@@ -1,21 +0,0 @@
-SELECT
```
Comment (alexey-milovidov) [inline tests/queries/0_stateless/01379_with_fill_several_columns.sql:1]: You have accidentially deleted our tests.

Comment (etah000) [reply tests/queries/0_stateless/01379_with_fill_several_columns.sql:1]: sorry, it is a merge problem, I'll correct that.

**Comment by filimonov** [comment]
Someday real (i.e. 'cheap') alters/deletes may be introduced. Setting is required.

**Comment by etah000** [comment]
@alexey-milovidov  I added two setting variables in "Settings.h", but I cannot find a suitable way to get their values in the method " ParserAlterQuery::parseImpl" from "ParserAlterQuery.cpp". Passing those two parameters all the way down  is a little clumsy and tedious. Do we have a global variable for "context" or "settings"? I cannot find any.

@filimonov  Could you give some rough time schedule for the cheaper "delete/update" support?  I am also interested in its implementation principel.

**Comment by etah000** [comment]
I want to add setting variables as you required, but for now, the clickhouse codes are still complicated for me, and I cannot find a suitable way to modify the codes.  I don't want the request causes too much trouble for you and me. Sorry.

**Comment by alexey-milovidov** [comment]
@etah000 Sorry for late answer:

The setting should not modify parsing behaviour.
The parser should be able to parse simple UPDATE/DELETE queries unconditionally into separate AST.

If the setting is enabled, this AST will be converted to AST for ALTER UPDATE/DELETE and then query will be processed as usual. We can add `InterpreterUpdateQuery`, `InterpreterDeleteQuery` for this purpose. It will be natural for future extensions.

If the setting is disabled, exception should be thrown.
